dashboard/app/controllers/api/v1/users_controller.rb

```diff
@@ -34,6 +34,17 @@ def post_ui_tip_dismissed
     render status: 200, json: {set: property_name}
   end
 
+  # POST /api/v1/users/<user_id>/accept_data_transfer_agreement
+  def accept_data_transfer_agreement
+    @user.properties[:data_transfer_agreement_accepted] = true
+    @user.properties[:data_transfer_agreement_accepted_at] = DateTime.now
+    @user.properties[:data_transfer_agreement_request_ip] = params[:data_transfer_agreement_request_ip]
```

Comment:
As mentioned, can we just get the IP address from the `request` at this point?

Reply:
I think you can get it from `request.env['REMOTE_ADDR']`.